attestation-service/attestation-service/src/lib.rs

```diff
@@ -77,6 +78,24 @@ pub enum Data {
     Structured(Value),
 }
 
+#[derive(Error, Debug)]
+pub enum ServiceError {
+    #[error("io error: {0}")]
+    IO(#[from] std::io::Error),
+    #[error("Parse error:{0}")]
+    ParseError(#[from] strum::ParseError),
+    #[error("Policy Engine {0} is not supported")]
+    UnsupportedPolicyEngine(String),
+    #[error("Create rvps failed.")]
+    Rvps(#[from] RvpsError),
+    #[error("Set Policy failed {0}")]
+    SetPolicyFailed(String),
+    #[error("Starting service failed: {0}")]
+    StartService(String),
+    #[error("Anyhow error: {0}")]
+    Anyhow(#[from] anyhow::Error)
```

Comment:
`anyhow` is too generic. Could you try to classify the source of this error as one of the above and use `map_err`?

A suggestion is to firstly define a set of meanful error types with determined semantics. Then Let's go to use them by `map_err`. The definition should not be too generic, otherwise the user will be confused if an error is raised.

wdyt?